### DISCUSSION TITLE
Need a better way to see when a member is finally saved.

### DISCUSSION THREAD

**Discussion by dhfreinkel** [codefori/vscode-ibmi#1054] [Q&A]
How do I see when a member is finally saved?
This is a big problem as sometimes its quick and other times its slow. With large source members, the updated to the server takes a while. 

Because we have old code where files are defined twice, once for inquiry and the second for updates. The 2 files are the same file but with a suffix to the file name. We have CL the does an OVRDBF of the dummy file name back to the file. So I have to compile outside of VSC.

Many times now I have started the compile to find the saving to disk has not completed and so the compile fails.

If I use ctrl + E, I can see the delay, but the programs never compile.

So I need something on the screen to tell me that the member has now been saved.

Or, is there something I am not seeing?

TIA, Darryl Freinkel.

**Comment by worksofliam** [comment]
> How do I see when a member is finally saved?

When a file is dirty (editor and not saved), then the tab will display a circle. When you hover over the tab is will turn into an X (to close it). The circle goes away when it is clean (saved with no changes pending).

There is also a setting (from the connection settings) that will enable auto save before running an Action: 

<img width="568" alt="image" src="https://user-images.githubusercontent.com/3708366/218230849-c230b2e0-8fe4-4ba6-964d-172a259ba073.png">

**Reply by Gregory1701** [reply]
I hadn't found this setting yet, but thanks @worksofliam, this is exactly what I was needing!  One interesting note, however, is that 'out of the box' I wasn't getting this dialog box when attempting an action on a dirty source member:
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/c031cf76-b588-467c-bf83-614febd71bdd)
Once I turn the auto save for actions setting on, tested it out, then turned it back off, I started getting the prompt dialog above. I clicked on the save automatically button which turned the setting back on for me, which is great.  I plan to leave this setting on.  However, on a fresh install you may want to have a look and see why it doesn't by default give the above prompt.  Maybe it was just something weird with my setup, but seems like something worth looking into.

Thanks again for the setting, I love it!

**Comment by chrjorgensen** [comment]
> Because we have old code where files are defined twice, once for inquiry and the second for updates. The 2 files are the same file but with a suffix to the file name. We have CL the does an OVRDBF of the dummy file name back to the file. So I have to compile outside of VSC.

As a sidenote, I would recommend looking at implementing the [BUILD tool](https://www.scottklement.com/build/) created years ago by @ScottKlement and very well [documented](https://www.scottklement.com/build/builddocs.html). I have used it for years (and adapted it to also work with streamfile source files), and it works great. It doesn't interfere with the regular compile commands (since it relies on comments in the source) and can eliminate many repetitious tasks.

That way you can compile (or rather BUILD) from VSC and get the compiler output back to VSC.

**Comment by m-tyler** [comment]
For those users that have the setting `Show SEU Colors` set on you will see slower performance when saving members.  Turn this feature off if you don't truly need to have SEU coloring dealt with in source editing.
![image](https://github.com/codefori/vscode-ibmi/assets/1353665/eb484a7d-52d8-4348-b19c-ed57bd5f3a03)
As the tool provides for good source coloring, I think SEU colors are not as useful anymore.

**Reply by worksofliam** [reply]
Maybe we should just remove this feature once-and-for-all 😁

**Reply by m-tyler** [reply]
Ehh🤔.  It certainly took me a little bit to understand what was going on.   I don't recall if that feature was turned on by default or by myself.   Maybe add a note somewhere if the user goes to activate that feature.   

I wrote a RPG program to remove color code characters from source I have out for modification so I am not as negatively affected.   

Maybe a `sed` string might be a faster option?

**Reply by worksofliam** [reply]
It is off by default! Plus we always recommend to remove the color hex.
